api/internal/localizer/localizer.go

```diff
@@ -75,9 +78,55 @@ func (lc *Localizer) Localize() error {
 	return nil
 }
 
-// TODO(annasong): implement
-// processKust returns a copy of the kustomization at kt with paths localized.
-func (lc *Localizer) processKust(kt *target.KustTarget) *types.Kustomization {
+// processKust returns a copy of the kustomization at kt with component paths localized.
+func (lc *Localizer) processKust(kt *target.KustTarget) (*types.Kustomization, error) {
 	kust := kt.Kustomization()
-	return &kust
+
+	for i, path := range kust.Components {
+		newPath, err := lc.localizeDir(path)
+		if err != nil {
+			return nil, errors.WrapPrefixf(err, "unable to localize components field")
+		}
+		kust.Components[i] = newPath
+	}
+
+	return &kust, nil
+}
+
+// localizeDir localizes root path and returns the localized path
+func (lc *Localizer) localizeDir(path string) (string, error) {
+	ldr, err := lc.ldr.New(path)
+	if err != nil {
+		return "", errors.Wrap(err)
+	}
+	defer func() { _ = ldr.Cleanup() }()
+
+	var locPath string
+	if repo := ldr.Repo(); repo != "" {
+		// TODO(annasong): check if able to add localize directory
+		locPath = locRootPath(path, filesys.ConfirmedDir(repo), filesys.ConfirmedDir(ldr.Root()))
+	} else {
+		locPath, err = filepath.Rel(lc.ldr.Root(), ldr.Root())
+		if err != nil {
+			//nolint:gocritic // should never occur, but if hit, something fundamentally wrong; should immediately exit
```

Comment:
what is the linter complaining about?

Reply:
If `log.Fatal` is hit, the `defer` statement earlier in this function isn't run. 

I think this is a sound check, but in our case, we only call `log.Fatal` if we hit an unexpected situation in which our existing assumptions are overthrown. In such a case, I think it's valid to immediately exit without trying to clean state? Open to other options as well.